infoblox/resource_infoblox_zone_auth.go

```diff
@@ -0,0 +1,179 @@
+package infoblox
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ibclient "github.com/infobloxopen/infoblox-go-client/v2"
+)
+
+func resourceZoneAuth() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceZoneAuthCreate,
+		Read:   resourceZoneAuthGet,
+		Update: resourceZoneAuthUpdate,
+		Delete: resourceZoneAuthDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"fqdn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The FQDN of the Authoritative zone",
+			},
+			"ns_group": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of Nameserver Group in Infoblox; will create NS records",
+			},
+			"restart_if_needed": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				Description: "Restart the member service if necessary for changes to take effect",
+			},
+			"comment": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Description of this Authoritative Zone; max 256 characters",
+			},
+			"soa_default_ttl": {
+				Type:        schema.TypeInt,
+				Default:     3600,
+				Optional:    true,
+				Description: "Time To Live (TTL) of the SOA record, in seconds",
+			},
+			"soa_expire": {
+				Type:        schema.TypeInt,
+				Default:     2419200,
+				Optional:    true,
+				Description: "Time in seconds for secondary servers to stop answering about the zone because the data is stale (default 1 week)",
+			},
+			"soa_negative_ttl": {
+				Type:        schema.TypeInt,
+				Default:     900,
+				Optional:    true,
+				Description: "Time in seconds for secondary servers to cache data for 'Does Not Respond' responses",
+			},
+			"soa_refresh": {
+				Type:        schema.TypeInt,
+				Default:     10800,
+				Optional:    true,
+				Description: "Interval in seconds for secondary servers to check the primary server for fresh data about the zone",
+			},
+			"soa_retry": {
+				Type:        schema.TypeInt,
+				Default:     3600,
+				Optional:    true,
+				Description: "Interval in seconds for secondary servers to wait before recontacting primary server about the zone after failure",
+			},
+			"ext_attrs": {
+				Type:        schema.TypeString,
+				Default:     "",
+				Optional:    true,
+				Description: "Extensible attributes, as a map in JSON format",
+			},
+		},
```

Comment:
There must be the field about DNS view, and preferably about zone format (forward or reverse, like in WAPI).

Reply:
I added these in the matching Go Client PR above and to the resource here in the provider